docs/training.md

```diff
@@ -0,0 +1,204 @@
+annotations_file = preprocess.split_raster(path_to_raster=raster,
+                                           annotations_file="{}/example.csv".format(tmpdir),
+                                           base_dir=tmpdir,
+                                           patch_size=500,
+                                           patch_overlap=0.25)
+
+# Returns a 6 column pandas array
+assert annotations_file.shape[1] == 6
+```
+
+Now we have crops and annotations in 500 px patches for training.
+
+### Negative samples
+
+To include images with no annotations from the target classes create a dummy row specifying the image_path, but set all bounding boxes to 0
+
+```
+image_path, xmin, ymin, xmax, ymax, label
+myimage.png, 0,0,0,0,"Tree"
+```
+
+Excessive use of negative samples may have negative impact on model performance, but used sparingly it can increase precision. These samples are removed from evaluation and do not count in precision/recall. 
+
+### Model checkpoints
+
+Pytorch lightning allows you to [save a model](https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#checkpoint-callback) at the end of each epoch. By default this behevaior is turned off. To restore model checkpointing
+
+```
+callback = ModelCheckpoint(dirpath='temp/dir',
+                                 monitor='box_recall', 
+                                 mode="max",
+                                 save_top_k=3,
+                                 filename="box_recall-{epoch:02d}-{box_recall:.2f}")
+model.create_trainer(logger=TensorBoardLogger(save_dir='logdir/'), 
+                                  callbacks=[callback])
+model.trainer.fit(model)
+```
+### Saving and loading models
+
+```
+import tempfile
+import pandas as pd
+tmpdir = tempfile.TemporaryDirectory()
+
+model.use_release()
+
+#save the prediction dataframe after training and compare with prediction after reload checkpoint 
+img_path = get_data("OSBS_029.png")
+model.create_trainer()
+model.trainer.fit(model)
+pred_after_train = model.predict_image(path = img_path)
+
+#Create a trainer to make a checkpoint
+model.trainer.save_checkpoint("{}/checkpoint.pl".format(tmpdir))
+
+#reload the checkpoint to model object
+after = main.deepforest.load_from_checkpoint("{}/checkpoint.pl".format(tmpdir))
+pred_after_reload = after.predict_image(path = img_path)
+
+assert not pred_after_train.empty
+assert not pred_after_reload.empty
+pd.testing.assert_frame_equal(pred_after_train,pred_after_reload)
+```
+
+---
+
+Note that when reloading models, you should carefully inspect the model parameters, such as the score_thresh and nms_thresh. These parameters are updated during model creation and the config file is not read when loading from checkpoint!
+It is best to be direct to specify after loading checkpoint. If you want to save hyperparameters, edit the deepforest_config.yml directly. This will allow the hyperparameters to be reloaded on deepforest.save_model().
```

Comment:
Can we rephrase this sentence. 

`It is best to be direct to specify after loading checkpoint. If you want to save hyperparameters, edit the deepforest_config.yml directly. This will allow the hyperparameters to be reloaded on deepforest.save_model().`